pkg/nvcdi/driver-nvml.go

```diff
@@ -102,17 +102,19 @@ func (l *nvcdilib) NewDriverLibraryDiscoverer(version string) (discover.Discover
 	driverDotSoSymlinksDiscoverer := discover.WithDriverDotSoSymlinks(
 		libraries,
 		version,
-		l.nvidiaCDIHookPath,
+		l.hookCreator,
 	)
 	discoverers = append(discoverers, driverDotSoSymlinksDiscoverer)
 
-	if l.HookIsSupported(HookEnableCudaCompat) {
-		// TODO: The following should use the version directly.
-		cudaCompatLibHookDiscoverer := discover.NewCUDACompatHookDiscoverer(l.logger, l.nvidiaCDIHookPath, l.driver)
+	// TODO: The following should use the version directly.
+
+	// replace l.nvidiaCDIHookPath for the l.cdiHook to use the inferface
+	if !l.disabledHooks[HookEnableCudaCompat] {
```

Comment:
```suggestion
	if l.HookIsSupported(HookEnableCudaCompat) {
```
It should be clear from the change set that this behaviour is NOT affected by this change.